gandalf.js

```diff
@@ -1,92 +1,95 @@
 /**
```

Comment:
SCRIPT-This function might benefit from using async/await instead of callbacks for better readability.